dissect/target/plugins/os/windows/tasks.py

```diff
@@ -58,452 +61,124 @@
-        task.logon_type = task.get_element("Principals/Principal/LogonType")
-        task.group_id = task.get_element("Principals/Principal/GroupId")
-        task.display_name = task.get_element("Principals/Principal/DisplayName")
-        task.run_level = task.get_element("Principals/Principal/RunLevel")
-        task.process_token_sid_type = task.get_element("Principals/Principal/ProcessTokenSidType")
-        task.required_privileges = task.get_element("Principals/Principal/RequiredPrivileges")
+        Yields:
+            The scheduled tasks found on the target.
+        """
+        for task_object in self.task_objects:
+            yield from self.get_task_fields(task_object)
 
-        # Settings
-        task.allow_start_on_demand = task.get_element("Settings/AllowStartOnDemand")
-        task.restart_on_failure_interval = task.get_element("Settings/RestartOnFailure/Interval")
-        task.restart_on_failure_count = task.get_element("Settings/RestartOnFailure/Count")
-        task.mutiple_instances_policy = task.get_element("Settings/MultipleInstancesPolicy")
-        task.dissalow_start_on_batteries = task.get_element("Settings/DisallowStartIfOnBatteries")
-        task.stop_going_on_batteries = task.get_element("Settings/StopIfGoingOnBatteries")
-        task.allow_hard_terminate = task.get_element("Settings/AllowHardTerminate")
-        task.start_when_available = task.get_element("Settings/StartWhenAvailable")
-        task.network_profile_name = task.get_element("Settings/NetworkProfileName")
-        task.run_only_network_available = task.get_element("Settings/RunOnlyIfNetworkAvailable")
-        task.wake_to_run = task.get_element("Settings/WakeToRun")
-        task.enabled = task.get_element("Settings/Enabled")
-        task.hidden = task.get_element("Settings/Hidden")
-        task.delete_expired_task_after = task.get_element("Settings/DeleteExpiredTaskAfter")
-        task.idle_duration = task.get_element("Settings/IdleSettings/Duration")
-        task.idle_wait_timeout = task.get_element("Settings/IdleSettings/WaitTimeout")
-        task.idle_stop_on_idle_end = task.get_element("Settings/IdleSettings/StopOnIdleEnd")
-        task.idle_restart_on_idle = task.get_element("Settings/IdleSettings/RestartOnIdle")
-        task.network_settings_name = task.get_element("Settings/NetworkSettings/Name")
-        task.network_settings_id = task.get_element("Settings/NetworkSettings/Id")
-        task.execution_time_limit = task.get_element("Settings/ExecutionTimeLimit")
-        task.priority = task.get_element("Settings/Priority")
-        task.run_only_idle = task.get_element("Settings/RunOnlyIfIdle")
-        task.unified_scheduling_engine = task.get_element("Settings/UseUnifiedSchedulingEngine")
-        task.disallow_start_on_remote_app_session = task.get_element("Settings/DisallowStartOnRemoteAppSession")
+    def get_task_fields(self, task_object) -> Iterator:
```

Comment:
Specify in the return type what it is an Iterator of.

This doesn't get tasks fileds but gets complete records, so the function name should also be something that describes that better.